lib/readline.js

```diff
@@ -481,6 +482,42 @@ Interface.prototype._insertString = function(c) {
     // A hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // Emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+// Append eager eval result to the
+// Current line
+Interface.prototype._appendPreview = function(result) {
```

Comment:
It would be possible to add internal functionality in a separate file that would only be loaded internally. That way we do not have to pollute the prototype further. An underscore in a property name would not hinder anyone from using or monkey patching this function.

Reply:
Agreed, I will move them to `internal/readline/util`.